cpp/oneapi/dal/test/macro.hpp

```diff
@@ -0,0 +1,114 @@
+/*******************************************************************************
```

Comment:
Do we really need this macros? Is there a way to do this using templates?

Reply:
No templates don't help here, generally macros are needed to simplify declaration of tests cases for algorithms, to iterate over all possible combinations of Floats and Methods, see `ALGO_TEST_CASE` (`dal/test/common.hpp`).

```c++
ALGO_TEST_CASE("test case name", (float, double), (method::m1, method::m2, method::m3)) {
   // This test case will run 2 x 3 = 6 times with the different Float and Method template parameters
   // For more details, have a look at Catch2 type parametrised test cases, this one is wrapper over `TEMPLATE_TEST_CASE`
}
```

Reply:
In other words, those macros are needed for template instantiation, you cannot **explicitly** instantiate template using the other template.

Reply:
Yeap. It was out of my scope. Thank you